src/CommandIsolator.cpp

```diff
@@ -82,6 +90,40 @@ process::Future<Option<ContainerLaunchInfo>> CommandIsolatorProcess::prepare(
   return containerLaunchInfo.get();
 }
 
+process::Future<ContainerLimitation> CommandIsolatorProcess::watch(
+    const ContainerID& containerId) {
+  if (m_watchCommand.isNone()) {
+    return process::Future<ContainerLimitation>();
+  }
+
+  logging::Metadata metadata = {containerId.value(), "watch"};
+
+  JSON::Object inputsJson;
+  inputsJson.values["container_id"] = JSON::protobuf(containerId);
+
+  Try<string> output = CommandRunner(m_isDebugMode, metadata)
+                           .run(m_watchCommand.get(), stringify(inputsJson));
+
+  if (output.isError()) {
+    LOG(WARNING) << "Unable to parse output: " << output.error();
+    return process::Future<ContainerLimitation>();
+  }
+
+  if (output->empty()) {
+    return process::Future<ContainerLimitation>();
+  }
+
+  Result<ContainerLimitation> containerLimitation =
+      jsonToProtobuf<ContainerLimitation>(output.get());
+
+  if (containerLimitation.isError()) {
+    LOG(WARNING) << "Unable to deserialize ContainerLimitation: "
+                 << containerLimitation.error();
+    return process::Future<ContainerLimitation>();
+  }
+  return containerLimitation.get();
```

Comment:
are you sure it matches the type Future<containerlimitation> ? it seems it returns a raw ContainerLimitation (I'm not expert on those types)